crimlog/github-projectv2-csv-exporter
# feat: support custom fields & field toggling

# Summary
- Resolves #1, #2
- Disregard all changes in `dist/`, it wasn't gitignored by the creator for some reason and I'm not going to modify that (just yet)
- Adds custom project fields to the CSV output generation
- Allows the user to configure which fields are displayed in the generated CSV

# Testing
- Clone repo, checkout `crimlog` branch
- `npm install`
- `npm start`
- Navigate to http://localhost:6006 in your browser and proceed to [Exporter Settings](http://localhost:6006/?path=/story/tools-github-project-exporter--settings)
	- Set an access token and login value of "crimlog"
- On the [Exporter Settings](http://localhost:6006/?path=/story/tools-github-project-exporter--settings) page, you should be able to view several different fields as the last setting, along with the ability to toggle them on/off 
- Run an actual export in a local environment and confirm the actual CSV output matches the expected result

# Imagery
Web application:
![image](https://user-images.githubusercontent.com/35435704/207775366-1abb5c6e-063f-40ef-8b4f-ff09f9475f50.png)

CSV output:
![image](https://user-images.githubusercontent.com/35435704/207772883-1877c3a6-62a1-4e4e-a836-3db2e52de852.png)
